### PR TITLE
Fix bad geth log decoding

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/eth/contracts/FluxAggregator_test.go [hunk 1]
```diff
@@ -249,3 +249,63 @@ func TestFluxAggregatorClient_LatestSubmission(t *testing.T) {
...
+
+	newRoundLogRaw := cltest.LogFromFixture(t, "../../services/testdata/new_round_log.json")
+	var newRoundLog contracts.LogNewRound
+	err = fa.UnpackLog(&newRoundLog, "NewRound", newRoundLogRaw)
+	require.NoError(t, err)
+	require.Equal(t, int64(1), newRoundLog.RoundId.Int64())
+	require.Equal(t, common.HexToAddress("f17f52151ebef6c7334fad080c5704d77216b732"), newRoundLog.StartedBy)
+	require.Equal(t, int64(15), newRoundLog.StartedAt.Int64())
+
+	type BadLogNewRound struct {
```
Comment (samsondav) [inline core/eth/contracts/FluxAggregator_test.go]: Why is it bad ?

File: core/eth/geth_copied.go [hunk 2]
```diff
@@ -47,6 +47,12 @@ func parseTopics(out interface{}, fields abi.Arguments, topics []common.Hash) er
 		if !arg.Indexed {
 			return errors.New("non-indexed field in topic reconstruction")
 		}
+
+		_, exists := reflect.TypeOf(out).Elem().FieldByName(capitalise(arg.Name))
```
Comment (samsondav) [inline core/eth/geth_copied.go]: I think we may need a comment on this.

Assuming it happens when the struct is out of sync with the log? When might that happen in real use?

**Comment by felder-cl** [comment]
Might be good to have a PR description and/or comment to help introduce context

File: core/eth/contracts/FluxAggregator.go [hunk 3]
```diff
@@ -41,16 +41,16 @@ type LogNewRound struct {
...
+	RestartDelay   uint32
+	Timeout        uint32
 	Address        common.Address
 }
 
 type LogAnswerUpdated struct {
 	Current   *big.Int
-	RoundID   *big.Int
+	RoundId   *big.Int
 	Timestamp *big.Int
```
Comment (felder-cl) [inline core/eth/contracts/FluxAggregator.go]: Maybe keep as RoundID?
https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: core/eth/contracts/FluxAggregator_test.go (hunk 1 above)
Comment (spooktheducks) [reply core/eth/contracts/FluxAggregator_test.go]: In Solidity, the round ID field is `roundId`.  Geth's Go code expects this to translate to `RoundId`, not `RoundID`, and fails to find the field if it doesn't.

File: core/eth/geth_copied.go (hunk 2 above)
Comment (spooktheducks) [reply core/eth/geth_copied.go]: Yeah, it's when the struct fields don't match what Geth's decoding logic expects.  This'll happen if we don't mind how we name fields when adding or updating them.

File: core/eth/contracts/FluxAggregator.go (hunk 3 above)
Comment (spooktheducks) [reply core/eth/contracts/FluxAggregator.go]: No, that was the whole issue in the first place -- Geth's code expected a lowercase final `d`

File: core/eth/contracts/FluxAggregator_test.go (hunk 1 above)
Comment (felder-cl) [reply core/eth/contracts/FluxAggregator_test.go]: https://github.com/ethereum/go-ethereum/pull/16648 <- does go-eth support these 'abi' tags?

Comment (spooktheducks) [reply core/eth/contracts/FluxAggregator_test.go]: Only on non-indexed fields.  I plan to rework their decoding logic in a subsequent PR to add this feature for indexed fields.

Comment (felder-cl) [reply core/eth/contracts/FluxAggregator_test.go]: Ok as long as it was considered

Comment (samsondav) [reply core/eth/contracts/FluxAggregator_test.go]: Can we pls have a comment explaining this. Not obvious.

**Pull request merged**